clang/include/clang/AST/DeclTemplate.h

```diff
@@ -301,7 +279,9 @@ class TemplateArgumentList final
   unsigned size() const { return NumArguments; }
 
   /// Retrieve a pointer to the template argument list.
-  const TemplateArgument *data() const { return Arguments; }
+  const TemplateArgument *data() const {
+    return getTrailingObjects<TemplateArgument>();
+  }
```

Comment:
I wonder if we could get rid of that entirely in the public interface

Reply:
We definitely could, but perhaps that is outside the scope of this PR? That being said, It would be good to unify the interfaces of `TemplateArgumentList`, `TemplateArgumentListInfo`, and `ASTTemplateArgumentListInfo` by perhaps giving them all the following members:
- `begin()` and `end()`
- `operator[]`
- `operator ArrayRef<TemplateArgument>`/`operator ArrayRef<TemplateArgumentInfo>`